hydra-tx/src/Hydra/Tx/Commit.hs

```diff
@@ -103,6 +104,19 @@ commitTx networkId scriptRegistry headId party commitBlueprintTx (initialInput,
       )
       (unRedeemers $ tx ^. witsTxL . rdmrsTxWitsL)
 
+  resolveNonSpendingRedeemers tx =
+    Map.foldMapWithKey
+      ( \p (d, ex) ->
+          case redeemerPointerInverse (tx ^. bodyTxL) p of
+            SJust (ConwayMinting (AsIxItem i _)) -> [(ConwayMinting (AsIx i), (d, ex))]
+            SJust (ConwayRewarding (AsIxItem i _)) -> [(ConwayRewarding (AsIx i), (d, ex))]
+            SJust (ConwayCertifying (AsIxItem i _)) -> [(ConwayCertifying (AsIx i), (d, ex))]
+            SJust (ConwayProposing (AsIxItem i _)) -> [(ConwayProposing (AsIx i), (d, ex))]
+            SJust (ConwayVoting (AsIxItem i _)) -> [(ConwayVoting (AsIx i), (d, ex))]
+            SJust (ConwaySpending (AsIxItem _ _)) -> []
+            SNothing -> []
+      )
+      (unRedeemers $ tx ^. witsTxL . rdmrsTxWitsL)
```

Comment:
Should: not call this "resolving" as it is actually filtering and can be done a lot simpler with something like
```suggestion
  nonSpendingRedeemers tx =
    Map.filterWithKey
      ( \p _ ->
          case p of
            ConwaySpending{} -> False
            _ -> True
      )
      (unRedeemers $ tx ^. witsTxL . rdmrsTxWitsL @LedgerEra)
```

Reply:
I agree with the name of the function but code changes would just yield true/false.